### PR TITLE
Refactors window construction a little, window construction steps are slightly faster

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Window construction steps are slightly faster; normal windows now take 6 seconds with standard tools, from 7 and reinforced windows now take 12 seconds with standard tools, from 14.
tweak: Brass windows take 8 seconds with standard tools, from 7.
/:cl:

That is, brass windows will take 8 with #24798; otherwise they'll take 16, but I'm not too worried about that.

### CODE REVIEW

File: code/game/objects/structures/window.dm [hunk 1]
```diff
@@ -155,59 +156,62 @@
...
 			return
 	return ..()
 
+/obj/structure/window/proc/check_state(checked_state)
+	if(state == checked_state)
+		return TRUE
+
+/obj/structure/window/proc/check_anchored(checked_anchored)
+	if(anchored == checked_anchored)
+		return TRUE
```
Comment (xxalpha) [inline code/game/objects/structures/window.dm]: but why

File: code/game/objects/structures/window.dm [hunk 2]
```diff
@@ -155,59 +156,62 @@
...
-
 				playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
 				user << "<span class='notice'>You successfully disassemble [src].</span>"
 				qdel(src)
 			return
 	return ..()
 
+/obj/structure/window/proc/check_state(checked_state)
+	if(state == checked_state)
+		return TRUE
```
Comment (xxalpha) [inline code/game/objects/structures/window.dm]: but why

File: code/game/objects/structures/window.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/objects/structures/window.dm]: Constant do_after checks

File: code/game/objects/structures/window.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/objects/structures/window.dm]: Constant do_after checks

File: code/game/objects/structures/window.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/structures/window.dm]: wish we had anonymous functions.

**Comment by Dorsisdwarf** [comment]
In this PR, Joan nerfs clock cult. We are entering a new era of tgstation

**Comment by ChangelingRain** [comment]
Uh, no. #24798 was the nerf, this is a very slight buff to clock cult.

**Comment by Dorsisdwarf** [comment]
I am now shame

File: code/game/objects/structures/window.dm (hunk 1 above)
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: I'm fixing this with obj construction, gimmie 5 more years

**Pull request merged**